cpu/stm32_common/periph/rtc.c

```diff
@@ -322,4 +329,257 @@ void ISR_NAME(void)
     cortexm_isr_end();
 }
 
+#elif defined(CPU_FAM_STM32F1)
+
+void rtc_init(void)
+{
+    DEBUG("[RTC rtc_init]\n");
+
+    // Enable APB1 clocks
+    RCC->APB1ENR |= RCC_APB1ENR_PWREN | RCC_APB1ENR_BKPEN;
+
+    // Disable backup domain write protection
+    PWR->CR |= PWR_CR_DBP;
+
+    if ((RCC->BDCR & RCC_BDCR_RTCEN) != RCC_BDCR_RTCEN) // if RTC clock disabled
```

Comment:
The idea here is to skip initialization if it already has happened, right?

Generally, RIOT drivers should not assume that hardware is in a defined/well behaving state. E.g. assume a watchdog timer triggered a reboot in the middle of a configuration change. In that case a full reconfiguration of that hardware is required during driver initialization.

Back to this PR: I think this implies that the state of the RTC can only be one of the following:

a) disabled
b) enabled and correctly configured

Lets say the RTC  is instead indeed enabled, but incorrectly configured. In that case the initialization would fail.

Reply:
After config RTC drop current time.
Now rtc_init save current time. Config run always.